lib/remap-lang/src/expression/path.rs

```diff
@@ -28,14 +28,31 @@ impl Path {
     pub(crate) fn new(segments: Vec<Vec<String>>) -> Self {
         Self { segments }
     }
+
+    pub fn segments(&self) -> &[Vec<String>] {
+        &self.segments
+    }
+
+    pub fn as_string(&self) -> String {
+        self.segments
+            .iter()
+            .map(|c| {
+                c.iter()
+                    .map(|p| p.replace(".", "\\."))
+                    .collect::<Vec<_>>()
+                    .join(".")
+            })
+            .collect::<Vec<_>>()
+            .join(".")
+    }
```

Comment:
I'll note that path-based lookup is still a bit awkward right now. I plan on overhauling this before 0.12 ships with Remap. This will likely also involve re-introducing the `Lookup` work that Ana worked on.

So until then, please ignore the awkwardness here (and in other places) when it comes to converting between Remap-based paths, and Vector event paths.